modules/lead/product-jenkins/dockercfg.tf

```diff
@@ -34,3 +34,31 @@ resource "kubernetes_secret" "jenkins_repository_dockercfg" {
 
   type = "Opaque"
 }
+
+resource "kubernetes_secret" "prod_image_pull_secrets" {
+  provider = kubernetes.production
+  metadata {
+    name      = "image-pull-secret"
+    namespace = module.product_base.production_namespace
+  }
+
+  type = "kubernetes.io/dockerconfigjson"
+
+  data = {
+    ".dockerconfigjson" = data.template_file.dockercfg.rendered
+  }
+}
+
+resource "kubernetes_secret" "staging_image_pull_secrets" {
+  provider = kubernetes.staging
+  metadata {
+    name      = "image-pull-secret"
```

Comment:
So to get his working we'll just have to have our deployment point to this new secret?

Reply:
that's the idea